appengine/genetics/js/visualization.js

```diff
@@ -226,319 +303,526 @@ Genetics.Visualization.resetChartData_ = function() {
+      // of the birth animation will be able to access it.
+      var offspring = Genetics.Visualization
+          .createMouseAvatar_(event['OPT_OFFSPRING']);
+      heartSrc = 'genetics/heart.png';
+    } else if (result == 'INCOMPATIBLE') {
+      message = getMouseName(proposingMouse) + ' mated with ' +
+          getMouseName(askedMouse) + ', another ' + proposingMouse.sex + '.';
+      heartSrc = 'genetics/rainbow-heart.png';
+    } else if (result == 'INFERTILE') {
+      message = 'Mating between ' + getMouseName(proposingMouse) +
+          ' and ' + getMouseName(askedMouse) + ' failed because ' +
+          getMouseName(askedMouse) + ' is sterile.';
+      heartSrc = 'genetics/grey-heart.png';
+    }
+
+    var mateResult = function() {
+      if (result == 'SUCCESS') {
+        Genetics.Visualization.animateAddMouse_(offspring, x, y, true,
+            goog.bind(offspring.freeMouse, offspring));
+      } else {
+        Genetics.log(message);
+      }
+      proposingMouse.freeMouse(Genetics.Visualization.wanderAfterMate);
+      askedMouse.freeMouse(Genetics.Visualization.wanderAfterMate);
+    };
+    proposingMouse.busy = true;
+    askedMouse.busy = true;
+    Genetics.Visualization.moveMiceTogether_(proposingMouse, askedMouse,
+        goog.partial(Genetics.Visualization.showImage_, heartSrc, x, y, 50, 700,
+            mateResult));
+  }
 };
 
+
 /**
- * Remove a mouse and update internal counts.
- * @param {!Genetics.Mouse} mouse
+ * Display the end game summary.
  * @private
  */
-Genetics.Visualization.removeMouse_ = function(mouse) {
-  Genetics.Visualization.mouseSexes_[mouse.sex] -= 1;
-  Genetics.Visualization.pickFightOwners_[mouse.pickFightOwner] -= 1;
-  Genetics.Visualization.proposeMateOwners_[mouse.proposeMateOwner] -= 1;
-  Genetics.Visualization.acceptMateOwners_[mouse.acceptMateOwner] -= 1;
-  delete Genetics.Visualization.MICE[mouse.id];
+Genetics.Visualization.displayGameEnd_ = function() {
+  Genetics.Visualization.stop();
+  if (Genetics.Visualization.gameRankings_) {
+    var functionWinnersText = {
+      'pickFight': '', 'proposeMate': '',
+      'acceptMate': ''
+    };
+    var mouseFunctions = ['pickFight', 'proposeMate', 'acceptMate'];
+    for (var i = 0; i < mouseFunctions.length; i++) {
+      var mouseFunction = mouseFunctions[i];
+      var functionWinnersList =
+          Genetics.Visualization.gameRankings_[mouseFunction][0];
+      for (var j = 0; functionWinnersList && j < functionWinnersList.length;
+          j++) {
+        var playerId = functionWinnersList[j];
+        var stats = Genetics.Visualization.playerStatsDivs_[playerId];
+        if (stats[mouseFunction + 'Div']) {
+          goog.dom.classlist.add(stats[mouseFunction + 'Div'],
+              'winningFunction');
+        }
+        functionWinnersText[mouseFunction] +=
+            Genetics.Cage.players[playerId].name + ' ';
+      }
+    }
+
+    // var endGameDiv = document.createElement('div');
```

Comment:
Unfinished end game popup. Removed for this pull request and put into branch for new pull request with documentation.